src/library/Button/__tests__/Button.spec.js

```diff
@@ -29,5 +44,87 @@ describe('Button', () => {
     expect(button.props().onClick).toHaveBeenCalled();
   });
 
+  describe('types', () => {
```

Comment:
Not sure that this is the best approach...  While we are probably over reliant on snapshots, this happens to be the perfect case for one.  All of this can likely be simplified considerably.